app/user_controller.go

```diff
@@ -0,0 +1,167 @@
+package app
+
+import (
+	"crypto/sha1"
+	"database/sql"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/Arijeet-webonise/gameReview/app/models"
+	"github.com/satori/go.uuid"
+)
+
+var dbUser = map[string]*models.User{}
+var dbSession = map[string]string{}
+
+func encrypt(text string) string {
+	h := sha1.New()
+
+	h.Write([]byte(text))
+
+	bs := h.Sum(nil)
+
+	return fmt.Sprintf("%x", bs)
+}
+
+func (app *App) SignUpRender(w http.ResponseWriter, r *http.Request) {
+	tmplList := []string{"./web/views/base.html",
+		"./web/views/header.html",
+		"./web/views/footer.html",
+		"./web/views/User/signup.html"}
+
+	res, err := app.TplParser.ParseTemplate(tmplList, nil)
+	if err != nil {
+		app.Log.Error(err)
+	}
+	io.WriteString(w, res)
+}
+
+func (app *App) CheckIfLogined(r *http.Request) error {
```

Comment:
no need for this method.This should be part of your `AuthenticationService` which is called from the middleware